src/terrain/clients/iplant_groups.clj

```diff
@@ -241,75 +245,115 @@
 ;; This function kind of uses a hack. A search string is required, but if we make it the
 ;; same as the folder name then that approximates listing all groups in the folder. An
 ;; update to iplant-groups will be required to eliminate this hack.
-(defn get-teams [user {:keys [search creator member]}]
+(defn- get-teams* [team-type user {:keys [search creator member]}]
   (let [client (get-client)
-        folder (get-team-folder-name client creator)]
+        folder (get-team-folder-name client team-type creator)]
     (->> (cond member (fn [_] (find-groups-with-member client user member folder search))
                search (partial c/find-groups client user search)
-               :else  (partial c/find-groups client user))
-         (get-teams* client user folder))))
+               :else (partial c/find-groups client user))
+         (find-teams* client team-type user folder))))
+
+(defn get-teams [user params]
+  (get-teams* group-type-teams user params))
 
-(defn- grant-initial-team-privileges [client user group public-privileges]
+(defn get-communities [user params]
+  (get-teams* group-type-communities user params))
+
+(defn- grant-initial-team-privileges [client user group initial-admin public-privileges]
   (c/update-group-privileges client user group
-                             {:updates [{:subject_id (config/grouper-user) :privileges ["admin"]}
+                             {:updates [{:subject_id initial-admin :privileges ["admin"]}
                                         {:subject_id c/public-user :privileges public-privileges}]}))
 
 (defn add-team [user {:keys [name description public_privileges] :or {public_privileges []}}]
   (let [client (get-client)
-        folder (get-team-folder-name client user)]
-    (ensure-team-folder-exists client user)
-    (let [full-name (str folder ":" name)
+        folder (get-team-folder-name client group-type-teams user)]
+    (ensure-team-folder-exists client group-type-teams user)
+    (let [full-name (full-group-name name folder)
           group     (c/add-group client user full-name team-group-type description)]
-      (grant-initial-team-privileges client user full-name public_privileges)
-      (format-group (get-team-folder-name client) group))))
+      (grant-initial-team-privileges client user full-name (config/grouper-user) public_privileges)
+      (format-group (get-team-folder-name client group-type-teams) group))))
 
-(defn get-team [user name]
+(defn add-community [user {:keys [name description public_privileges] :or {public_privileges []}}]
   (let [client (get-client)
-        folder (get-team-folder-name client)]
+        folder (get-team-folder-name client group-type-communities)]
+    (ensure-folder-exists client (config/grouper-user) folder)
+    (let [full-name (full-group-name name folder)
+          group     (c/add-group client (config/grouper-user) full-name team-group-type description)]
```

Comment:
I just tried it, and apparently the groups service will return a `500 Server Error`:
```
{
  "error_code": "ERR_REQUEST_FAILED",
  "grouper_result_code": "PROBLEM_SAVING_GROUPS",
  "grouper_result_message": "There were 0 successes and 1 failures of saving groups."
}
```

Reply:
Ah, okay. So it'll prevent creating it, but only via an obscure Grouper error, heh. I don't know if it's worth checking whether a community of a given name already exists and throwing a more descriptive error in that case.

Reply:
That's probably a good idea.

Since the same issue exists for teams (though it would have to be a user duplicating one of their own teams), I think I'll save the fix for both endpoints for another PR.